engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeCompiler.scala

```diff
@@ -125,6 +128,40 @@ class NodeCompiler(definitions: ProcessDefinition[ObjectWithMethodDef],
     NodeCompilationResult(expressionTypingInfo, None, newCtx, validParams)
   }
 
+  def compileExpression(expr: graph.expression.Expression,
+                        outputVarName: String,
+                        ctx: ValidationContext)
+                        (implicit nodeId: NodeId): NodeCompilationResult[expression.Expression] = {
+    val (expressionTypingResult, validatedExpression) = validateExpression(expr, DefaultExpressionId, ctx, typing.Unknown)
+    val (newCtx, combinedValidatedExpression) = withVariableCombined(ctx, outputVarName, expressionTypingResult.typingResult, validatedExpression)
```

Comment:
I think we can get rid of withVariableCombined, and just Validated[ValidationContext] in NodeCompilationResult?